ee/app/models/server/raw/LivechatDepartmentAgents.ts

```diff
@@ -20,7 +20,7 @@ export class LivechatDepartmentAgentsRaw extends Raw {
 				preserveNullAndEmptyArrays: true,
 			},
 		};
-		const withBusinessHourId = { $match: { 'departments.businessHourId': { $exists: true } } };
+		const withBusinessHourId = { $match: { 'departments.businessHourId': businessHourId } };
```

Comment:
I think there is no index for the field `businessHourId`, right?
If I'm right, we should add this index in order to improve the query processes.